packages/typescript-estree/src/create-program/shared.ts

```diff
@@ -38,11 +38,41 @@ function canonicalDirname(p: CanonicalPath): CanonicalPath {
   return path.dirname(p) as CanonicalPath;
 }
 
+function getScriptKind(
+  extra: Extra,
+  filePath: string = extra.filePath,
+): ts.ScriptKind {
+  const extension = path.extname(filePath);
+  // note - we respect the user's extension when it is known  we could override it and force it to match their
+  // jsx setting, but that could create weird situations where we throw parse errors when TSC doesn't
+  switch (extension.toLowerCase()) {
+    case 'ts':
+      return ts.ScriptKind.TS;
+
+    case 'tsx':
+      return ts.ScriptKind.TSX;
+
+    case 'js':
+      return ts.ScriptKind.JS;
```

Comment:
It is very common to use JSX in plain `.js` files as far as I know.

Reply:
Typescript does not ban JSX within plain JS files - you can provide JSX in both `.js` and `.jsx` files.

It does ofc enforce it for TS files - you can only provide JSX in `.tsx` files.

![image](https://user-images.githubusercontent.com/7462525/67361307-11066800-f51d-11e9-8016-33097c995b63.png)

Reply:
Cool, I'm so familiar with `ts.ScriptKind` then. Thx for your great job.